src/save.rs

```diff
@@ -0,0 +1,43 @@
+use anyhow::{format_err, Result};
+use serde::{de::DeserializeOwned, Serialize};
+
+use std::fs;
+use std::path::PathBuf;
+
+pub trait SaveData: DeserializeOwned + Serialize + Default {
+    fn relative_path() -> PathBuf;
+
+    fn path() -> Result<PathBuf> {
+        let mut data_dir =
```

Comment:
Could we put a conditional here: #[cfg(debug_assertions)]
Eg. when debug (dev dir) just write to the project root. Otherwise write to data dir. This way developing and actually using the software won't interfere? Check how it's done in main.rs for the log dir ;)

Reply:
An idea might be to make the logging initiation and this fetch their root dir from the same method so they share the logic. That's probably the best way to do it. Then we can re-use it for future file io.

Reply:
Yup, I like this idea. We can use `lazy_static` to define a global variable at runtime (needed for `dirs`) and mix in the conditional for debug assertions to use the crate root.